lib/util/transformer/length-based-splitter.js

```diff
@@ -96,6 +118,42 @@ export class LengthBasedSplitter extends Stream.Transform {
     this.isShutdown = true;
   }
 
+  /**
+   * Log information about large frames for debugging purposes
+   * @param {number} messageLength - The size of the detected frame
+   */
+  _logLargeFrame(messageLength) {
+    log.warn(`Large frame detected: ${util.toReadableSizeString(messageLength)}`);
+    log.warn(`Service: ${this.serviceName}, Max allowed: ${util.toReadableSizeString(this.maxFrameLength)}`);
+    
+    // Log memory usage if monitoring is enabled
+    if (DEBUG_CONFIG.enableMemoryMonitoring) {
+      const memUsage = process.memoryUsage();
+      log.warn(`Memory usage - RSS: ${util.toReadableSizeString(memUsage.rss)}, Heap: ${util.toReadableSizeString(memUsage.heapUsed)}`);
+    }
+  }
+
+  /**
+   * Create frame size error with enhanced debugging information
+   * @param {number} messageLength - The size of the frame that exceeded the limit
+   * @returns {Error} The constructed error object
+   */
+  _makeFrameSizeError(messageLength) {
+    const errorMsg = `The frame is bigger than expected. Length: ${util.toReadableSizeString(messageLength)}, max: ${util.toReadableSizeString(this.maxFrameLength)}`;
+    log.error(errorMsg);
+    log.error(`Service: ${this.serviceName}, Frame offset: ${this.lengthFieldOffset}, Frame length: ${this.lengthFieldLength}`);
+    
+    // Only suggest debugging steps if debug mode is not already enabled
+    if (!DEBUG_CONFIG.enableFrameSizeLogging) {
+      log.error('To debug this issue, set environment variables:');
```

Comment:
you may also simply provide the link to the above document here instead of listing env variable names